website/content/en/preview/concepts/setup.md

```diff
@@ -0,0 +1,499 @@
+```
+The role created here includes several AWS managed policies, which are designed to provide permissions for specific uses needed by Karpenter to create and manage nodes.
+These include:
+
+* [AmazonEKS_CNI_Policy](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonEKS_CNI_Policy.html): Provides permission Amazon VPC CNI Plugin needs to configure EKS worker nodes.
+* [AmazonEKSWorkerNodePolicy](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonEKSWorkerNodePolicy.html): Lets Amazon EKS worker nodes connect to EKS Clusters.
+* [AmazonEC2ContainerRegistryReadOnly](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonEC2ContainerRegistryReadOnly.html): Allows access to repositories in the Amazon EC2 Container Registry.
+* [AmazonSSMManagedInstanceCore](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonSSMManagedInstanceCore.html): Adds AWS Systems Manager service core functions for Amazon EC2.
+
+
+## KarpenterControllerPolicy
+
+This section sets the permissions that the Karpenter Controller has to create and manage EC2 resources.
+Because the scope of the KarpenterControllerPolicy is an AWS region, the cluster's AWS region is included in the AllowScopedEC2InstanceActions.
+
+A KarpenterControllerPolicy sets the name of the policy, then defines of a set of resources and actions allowed for those resources.
+For our example, the KarpenterControllerPolicy would be named: `KarpenterControllerPolicy-joe-karpenter-demo`
+```
+  KarpenterControllerPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      ManagedPolicyName: !Sub "KarpenterControllerPolicy-${ClusterName}"
+      # The PolicyDocument must be in JSON string format because we use a StringEquals condition that uses an interpolated
+      # value in one of its key parameters which isn't natively supported by CloudFormation
+      PolicyDocument: !Sub |
+        {
+          "Version": "2012-10-17",
+          "Statement": [
+```
+
+### AllowScopedEC2InstanceActions
+
+The AllowScopedEC2InstanceActions statement ID (Sid) identifies a set of EC2 resources that are allowed to be used with
+[RunInstances](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html) and [CreateFleet](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateFleet.html) actions.
+
+```
+            {
+              "Sid": "AllowScopedEC2InstanceActions",
+              "Effect": "Allow",
+              "Resource": [
+                "arn:${AWS::Partition}:ec2:${AWS::Region}::image/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}::snapshot/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:*:spot-instances-request/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:*:security-group/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:*:subnet/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:*:launch-template/*"
```

Comment:
How come this isn't locked down further (eg using tags and tag conditions)? It overrides a more specific statement later in the template.

Reply:
I'm not sure. I'll pass that information on to the engineering team.

Reply:
AFAIK, there isn't a way to lock down calls to `RunInstances` and `CreateFleet` for these resources further without knowing more information about those resources themselves.

For instance, `ec2:RunInstances` will use an image in your account as part of the instance launch. Getting access to that instance requires permission on that image, but Karpenter wasn't involved in creating that image and, therefore, has no specific knowledge about what the properties of that image are at all. A proper thing to do for most users would be to lock down the resources that Karpenter can interact with here by tagging these resources in a custom way and then using a `ResourceTag` condition to ensure that Karpenter only uses the resources that it has specific access to use

Reply:
> A proper thing to do for most users would be to lock down the resources that Karpenter can interact with here by tagging these resources in a custom way and then using a ResourceTag condition to ensure that Karpenter only uses the resources that it has specific access to use.

Yep, I usually do this. You can also restrict by VPC, by the tag that the launch template has, and more.

An AWS security specialist can explain more about the options here (also, @jonathan-innis, feel free to message me on Kubernetes Slack to follow up directly).

Reply:
I have updated to the latest cloudformation.yaml file to be used for v1beta1 since these comments entered. Please check to see if the updated content satisfies these concenrs.